superbench/analyzer/data_diagnosis.py

```diff
@@ -0,0 +1,234 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""A module for data analysis."""
+
+import re
+from typing import Callable
+
+import pandas as pd
+
+from superbench.common.utils import logger
+from superbench.analyzer.diagnosis_rule_op import RuleOp, DiagnosisRuleType
+import superbench.analyzer.file_handler as file_handler
+
+
+class DataDiagnosis():
+    """The DataDiagnosis class to do the baseline-based data diagnosis."""
+    def __init__(self):
+        """Init function using the raw data file path."""
+        self._sb_rules = {}
+        self._metrics = {}
+
+    def _get_metrics_by_benchmarks(self, metrics_list):
+        """Get all metrics by benchmark from metrics_list."""
+        benchmarks_metrics = {}
+        for metric in metrics_list:
+            benchmark = metric.split('/')[0]
+            if benchmark not in benchmarks_metrics:
+                benchmarks_metrics[benchmark] = set()
+            benchmarks_metrics[benchmark].add(metric)
+        return benchmarks_metrics
+
+    def _check_rules(self, rule, name):
+        """Check the rule of the metric whether formart is valid.
+
+        Args:
+            rule (dict): the rule
+            name ([type]): the rule name
+
+        Returns:
+            dict: criteria and rule for the metric
+        """
+        # check if rule is supported
+        if not isinstance(DiagnosisRuleType(rule['function']), DiagnosisRuleType):
```

Comment:
Will the constructor `DiagnosisRuleType(rule['function'])` fail before executing `isinstance`?

Reply:
it will raise exception, that's what I want